packages/web-extension/src/walletManager/walletManager.types.ts

```diff
@@ -46,11 +53,28 @@ export interface WalletManagerActivateProps<P extends string | number = string,
 }
 
 export interface WalletManagerApi {
+  chainId$: Observable<Cardano.ChainId>;
+
   /**
    * Create and activate a new ObservableWallet.
    * Reuses the store if the wallet was previously deactivated but not destroyed.
    */
   activate(props: WalletManagerActivateProps): Promise<void>;
+
+  /**
+   * Returns true if the wallet is active.
+   *
+   * @param props The properties of the wallet to check.
+   */
+  isActive(props: WalletManagerActivateProps): Promise<boolean>;
+
+  /**
+   * Switches the network of the active wallet.
+   *
+   * @param id The chain id of the network to switch to.
+   */
+  switchNetwork(id: Cardano.ChainId): Promise<void>;
+
```

Comment:
It wouldn't let me add a comment for `destroy()` as there were no changes. Let's rename it to `destroyData(walletId: WalletId)`. It shouldn't deactivate, user/app can call it separately. It can throw if you try to destroy wallet that is currently active.

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1012/commits/813029a3123dac04ffb0f783c6e239a3c19ae8e9